lib/appRelationsDiscovery.js

```diff
@@ -0,0 +1,62 @@
+const {validate} = require('./utils');
+
+const supportedSyntax = ['default','mermaid'];
+
+const defaultOptions = {
+  syntax: 'default',
+};
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
+async function getRelations(asyncApiDocs, { syntax } = defaultOptions) {
+  if (!Array.isArray(asyncApiDocs)) throw new Error('You must pass an array of AsyncApiDocuments on which you wish to discover the relations between');
+  if (!supportedSyntax.includes(syntax)) throw new Error('The given syntax is not supported');
+
+  const parsedAsyncApiDocs = await validate(asyncApiDocs);
```

Comment:
Please make here `try ... catch` for error handling :)

Reply:
Oops! missed this. will fix